### ISSUE TITLE
IoBroker-Überlastung führte zu falschen Tageswerten

### ISSUE THREAD

**Issue by w0nk1** [DrozmotiX/ioBroker.sourceanalytix#525]
Ich habe das Problem, dass vorhin ein JS den kompletten IoBroker lahmgelegt hatte. Nach dem Neustart des Rasperry Pi's schien auch erst wieder alles normal zu laufen.

Durch Zufall habe ich dann aber gesehen, dass **_current_day_** plötzlich einen utopisch falschen Wert liefert (Siehe Screenshot).

Über die Troubleshooting-Sektion werde ich leider nicht schlau, wie ich diesen Fehler wieder beheben bzw. geraderücken kann. 

Kann ich dazu beitragen dem Adapter in solchen Fällen mehr Resilenz zu verpassen?

: v0.4.14
 - JS-Controller version: 4.0.23
 - Node version: v16.15.1
 - Operating system: Raspbian GNU/Linux 10 (buster)

**Comment by DutchmanNL** [comment]
> Kann ich dazu beitragen dem Adapter in solchen Fällen mehr Resilenz zu verpassen?
> 

Ja! Würde sehr gerne die log Daten von dem Zeitpunkt haben und analysieren 😁

**Comment by w0nk1** [comment]
> > Kann ich dazu beitragen dem Adapter in solchen Fällen mehr Resilenz zu verpassen?
> 
> Ja! Würde sehr gerne die log Daten von dem Zeitpunkt haben und analysieren 😁

Da war der Adapter natürlich nicht im Debug. ;-) 

Ab heute zählt er auch wieder ganz normal für den aktuellen Tag. Könntest du mir sagen, wie ich den letzten Tag manuell eintragen kann? Weil so ist natürlich alles durcheinander von den kumulierten Werten. ^^

**Comment by DutchmanNL** [comment]
> Da war der Adapter natürlich nicht im Debug. ;-)

ist egal, was als info gekommen ist reicht dazu ;)

**Comment by w0nk1** [comment]
Ich befürchte das dich das dir das nicht viel hilft. Hier der Auszug aus dem Log, welche den Adapter betreffen:

> 2022-06-24 09:08:24.127  - [32minfo[39m: sourceanalytix.0 (4743) SourceAnalytix stopped, now you have to calculate by yourself :'( ...
2022-06-24 09:08:24.737  - [32minfo[39m: sourceanalytix.0 (4743) Terminated (START_IMMEDIATELY_AFTER_STOP): Without reason
2022-06-24 09:12:44.275  - [32minfo[39m: host.raspberrypi instance system.adapter.sourceanalytix.0 started with pid 4665
2022-06-24 09:13:05.638  - [32minfo[39m: sourceanalytix.0 (4665) starting. Version 0.4.14 in /opt/iobroker/node_modules/iobroker.sourceanalytix, node: v16.15.1, js-controller: 4.0.23
2022-06-24 09:13:05.822  - [32minfo[39m: sourceanalytix.0 (4665) Welcome to SourceAnalytix, making things ready ... 
2022-06-24 09:13:06.458  - [32minfo[39m: sourceanalytix.0 (4665) Found 1 SourceAnalytix enabled states
2022-06-24 09:13:06.459  - [32minfo[39m: sourceanalytix.0 (4665) Initialising (1 of 1) "hm-rpc.1.SEQ3473776.1.ENERGY_COUNTER"
2022-06-24 09:13:34.191  - [32minfo[39m: sourceanalytix.0 (4665) Initialization of hm-rpc.1.SEQ3473776.1.ENERGY_COUNTER successfully
2022-06-24 09:13:34.223  - [32minfo[39m: sourceanalytix.0 (4665) Successfully activated SourceAnalytix for 1 of 1 states, will do my Job until you stop me!


Das war ab dem Zeitpunkt des `sudo reboots`. Vorher gibt es keine Meldungen des Adapters.

**Comment by DutchmanNL** [comment]
@w0nk1 ich sehe im log nichts was eine Werte reset oder dergleichen gereichert hat. Welche Version setzt du ein?

Hatte sich der Wert des HomeMatic Geräts resettet auf 0?

**Comment by w0nk1** [comment]
v0.4.14

Mh, das schaut mir etwas verdächtig aus:

> 2022-06-24 00:00:00.114  - [32minfo[39m: sourceanalytix.0 (4743) Reset start values for : hm-rpc.1.SEQ3473776.1.ENERGY_COUNTER
2022-06-24 00:00:00.140  - [32minfo[39m: sourceanalytix.0 (4743) Memory values before reset : {"stateDetails":{"alias":"Stromverbrauch Wohnung","consumption":true,"costs":true,"deviceName":"hm-rpc__1__SEQ3473776__1__ENERGY_COUNTER","financialCategory":"costs","headCategory":"consumed","meter_values":true,"name":"Stromverbrauch Wohnung","stateType":"ElectricityDay","stateUnit":"Wh","useUnit":"kWh","deviceResetLogicEnabled":true,"threshold":1},"calcValues":{"start_day":803.5573,"start_month":396.2959,"start_quarter":0,"start_week":736.4959,"start_year":0,"valueAtDeviceInit":null,"valueAtDeviceReset":0,"cumulativeValue":824.2479000000001},"prices":{"basicPrice":"10,741","unitPrice":"0.26"}}

**Comment by DutchmanNL** [comment]
Joa dein Zähler ist auf < als voriger  gesprungen und es wurde ein Device reset ausgeführt

**Comment by w0nk1** [comment]
Passiert das immer wieder, bzw. ist das eine Eigenart des Homematic Energiezählers?  Weil eine Option dafür kann ich nirgends finden.

**Comment by DutchmanNL** [comment]
@w0nk1 jup passiert bei Stromlos machen, frage währe dan eher was danach nicht stimmt den wen der Wert z.b. 0 war sollte SA kumulieren (das sieht man auch im log) und damit wieder das korrekte total haben

wie lautet den der Wert vom HomeMatic DP, ist de wirklich nach 0 gegangen oder nur ein bissl unter den bekannten kumulierten Wert ?

**Comment by w0nk1** [comment]
![Bildschirmfoto 2022-06-25 um 12 08 35](https://user-images.githubusercontent.com/6522814/175769029-a5ff457a-e70b-4079-b1bf-3f0abb7f8ee4.png)

Das ist der besagte Tag. Das schaut um 0 Uhr eigentlich noch in Ordnung aus. Um 13 Uhr als der Fehler schon da war, läut dann irgedwas schief. Kann aber sein, dass ich das dann bei meinen kläglichen Versuchen der Fehlerbehebung dies selbst auslöste.

Aber der Fehler müsste ja schon um 0 Uhr seinen Ursprung haben?

**Comment by DutchmanNL** [comment]
Laut log ist das nachts passiert ja

**Comment by w0nk1** [comment]
Seltsam, die Werte scheinen da ja noch komplett in Ordnung zu sein. Wie bekomme ich denn nun den letzten Tag wieder manuell korrekt gesetzt? Ich blicke da nicht durch ^^

**Comment by DutchmanNL** [comment]
> Seltsam, die Werte scheinen da ja noch komplett in Ordnung zu sein. Wie bekomme ich denn nun den letzten Tag wieder manuell korrekt gesetzt? Ich blicke da nicht durch ^^

schau dafuer mal bitte im forum vorbei, hier ist ein post wo Crunchip r erklärt hat (ich muss docu und Funktionen dazu verbessern)

https://forum.iobroker.net/topic/40932/sourceanalytix-0-4-8-final-released/1091?_=1656159954765

ich schaffe es leider nicht fuer jedermann alles im 1:1 support zu erkläre aber zum Glück gibt es da viele Helfer im forum :)

**Comment by w0nk1** [comment]
Alles klar, verstehe ich. Aber das gehört definitiv noch irgendwie in die Einstellungen.  Blickt kein normalsterblicher durch. :-)

**Comment by DutchmanNL** [comment]
> Alles klar, verstehe ich. Aber das gehört definitiv noch irgendwie in die Einstellungen. Blickt kein normalsterblicher durch. :-)

Stimme ich zu, working in it Frontend ist nur nicht meine allergrößte Stärke 😥

**Comment by w0nk1** [comment]
Vllt .gibt es da Anregungen im Forum, wie man das lösen könnte

**Comment by DutchmanNL** [comment]
> Vllt .gibt es da Anregungen im Forum, wie man das lösen könnte

Anregungen genug, aber Ideen machen keine Code es muss auch umgesetzt werden 😉

**Comment by DutchmanNL** [comment]
Genau dieses Szenario — ioBroker steht oder ist überlastet, und danach stimmen die Tageswerte nicht mehr — sollte sich mit [v0.5.1](https://github.com/DrozmotiX/ioBroker.sourceanalytix/releases/tag/v0.5.1) deutlich anders verhalten.

SourceAnalytix merkt sich jetzt pro Quelle die zuletzt erfolgreich verarbeiteten Kalenderperioden und holt verpasste Tages-, Wochen-, Monats-, Quartals- und Jahreswechsel beim nächsten Start nach. Die Nachholung ist idempotent, ein Neustart kann dieselbe Periode also nicht doppelt archivieren oder zurücksetzen.

Ergänzend werden seit [v0.5.0](https://github.com/DrozmotiX/ioBroker.sourceanalytix/releases/tag/v0.5.0) Rückwärtssprünge des Zählers sauber klassifiziert (Reset, Messrauschen oder echter Rückgang), statt als utopischer Verbrauch durchzuschlagen.

Bitte einmal mit [v0.5.2](https://github.com/DrozmotiX/ioBroker.sourceanalytix/releases/tag/v0.5.2) gegenprüfen.

Wenn bis zum **11.08.2026** keine Rückmeldung kommt, schließe ich das Issue als veraltet. Es kann jederzeit wieder geöffnet werden.
